src/Data/List/ZipList.purs

```diff
@@ -25,6 +28,24 @@ newtype ZipList a = ZipList (List a)
 runZipList :: forall a. ZipList a -> List a
 runZipList (ZipList xs) = xs
 
+-- | Transpose any structure which is both `Foldable` and `Unfoldable`,
+-- | via `ZipList`; note that `sequence` for `ZipList` is the same as
+-- | `transpose`.
+transpose :: forall f a. (Foldable f, Unfoldable f) => f (f a) -> f (f a)
+transpose =
```

Comment:
In my opinion, such a highly generalized implementation as the above `transpose` does not belong here, because it has no dependency on `Data.List`. (One can argue it belongs in `Foldable` or `Unfoldable`.)